fix: restore pd-claude-opus-4-7 routing for crush/pi/omp/xcsh

## Summary

The post-a7bbbd7 upgrade to default `pd-claude-opus-4-7` broke every AI assistant in the container except Claude Code. Users reported `400 Invalid model name passed in model=claude-opus-4-7` from `crush`, `pi`, and `xcsh`.

## Root cause

`pd-claude-opus-4-7` is a custom F5 liteLLM proxy model ID that is not in pi-ai's or crush's built-in Anthropic catalog, so their resolvers strip the `pd-` prefix and emit `claude-opus-4-7`, which the proxy rejects.

## Reproduction (confirmed in-container)

- `curl -sS -X POST \"$LITELLM_BASE_URL/anthropic/v1/messages\" -H \"x-api-key: $LITELLM_API_KEY\" -H 'content-type: application/json' -H 'anthropic-version: 2023-06-01' -d '{\"model\":\"claude-opus-4-7\",\"max_tokens\":8,\"messages\":[{\"role\":\"user\",\"content\":\"hi\"}]}'` → `400 Invalid model name`
- Same curl with `pd-claude-opus-4-7` → `200`

## Fix

1. **`entrypoint.sh`** — extend pi/omp/xcsh `models.json` templating to register `pd-claude-opus-4-7` as a first-class custom model under the `anthropic` provider (pi-ai `ModelDefinitionSchema` compliant: id/api/reasoning/input/cost/contextWindow/maxTokens). Overwrite `~/.pi/agent/settings.json` every boot to defeat pi-ai's in-place fallback that strips `pd-` and persists the wrong `defaultModel`. Add `sed` substitution of `__CRUSH_BASE_URL__` → `${LITELLM_BASE_URL}/anthropic` for `crush.json`. Add a non-fatal startup curl probe that warns if the proxy returns non-200 for `pd-claude-opus-4-7`.
2. **`crush-config/crush.json`** — rewrite with `options.disable_provider_auto_update: true` (stops crush re-fetching the Catwalk catalog and wiping custom entries) plus an inline `providers.anthropic.models[]` declaration of `pd-claude-opus-4-7`. Use `__CRUSH_BASE_URL__` placeholder so no F5 hostname is committed. Point both large and small at `pd-claude-opus-4-7` (crush's embedded catalog aliases `claude-haiku-4-5` → `claude-haiku-4-5-20251001` which the proxy rejects; documented as follow-up).
3. **`opencode-config/opencode.json`** — fix stale `\"name\": \"Claude Opus 4.6\"` on the `pd-claude-opus-4-7` entry.

## Verification (post-fix, this container)

- `pi --print --provider anthropic --model pd-claude-opus-4-7 --no-session 'hi'` → response, no 400
- `xcsh -p --model pd-claude-opus-4-7 --no-session 'hi'` → response, no 400
- `crush run 'hi'` → response; logs show no 400 errors on subsequent runs
- `opencode run -m anthropic-proxy/pd-claude-opus-4-7 'hi'` → response
- `hermes chat -Q -q 'hi'` → response (OpenAI pass-through was never broken; included for completeness)
- `~/.pi/agent/settings.json` `defaultModel` remains `pd-claude-opus-4-7` after pi runs — no fallback rewrite
- `bash -n entrypoint.sh` passes; `jq .` validates both JSON files; `sed` renders a correct absolute URL

## Constraints honored

- No internal hostnames in tracked files (uses `__CRUSH_BASE_URL__` placeholder, same pattern as existing `__HERMES_BASE_URL__`/`__CODEX_BASE_URL__`)
- `entrypoint.sh` still consumes only `LITELLM_BASE_URL` and `LITELLM_API_KEY` from `.env` and derives everything else internally
- Model metadata values (1M context, 128k max output, pricing, cache) verified against Anthropic API docs for Opus 4.7 released 2026-04-16

## Known follow-ups (tracked separately after this lands)

- `@oh-my-pi/pi-natives` arm64 native-addon build failure: `undefined symbol: tree_sitter_glimmer_external_scanner_create`. Blocks `omp` start independent of this fix.
- Charmbracelet/crush: custom `providers.<name>.models[]` entries don't override embedded short-id aliases (so `claude-haiku-4-5` becomes `-20251001` and fails). Workaround is double-binding `small = large`; needs upstream issue on `charmbracelet/crush`.

## CI Failure Report

**PR:** #987
**Run:** 24612538281 (Super-Linter)
**Workflow:** lint / Lint Code Base

### Failed Checks

- `--> Linted: SHELL_SHFMT` — ERROR
- `lint / Lint Code Base` — FAILURE (consequence)

All other linters passed (EDITORCONFIG, GITHUB_ACTIONS, BASH_EXEC, BIOME_FORMAT, BIOME_LINT, CHECKOV, DOCKERFILE_HADOLINT, ENV, BASH, GITHUB_ACTIONS_ZIZMOR, GITLEAKS, GIT_MERGE_CONFLICT_MARKERS, JSCPD, LUA, MARKDOWN, NATURAL_LANGUAGE, SPELL_CODESPELL, TRIVY, YAML).

Local `pre-commit` gate also passed — the container's shfmt shell hook did not reject the file, but the super-linter upstream shfmt did. The devcontainer's pre-commit shell linter and super-linter's shfmt appear to disagree on `$(cat <<EOF ... EOF)` style.

### Diff demanded by super-linter shfmt

shfmt wants the two new `$(cat <<EOF ... EOF)` command substitutions in `entrypoint.sh` split onto their own lines:

```diff
-  _pi_models=$(cat <<EOF
+  _pi_models=$(
+    cat <<EOF
 {
   "providers": {
     ...
 }
 EOF
-)
+  )
```

Same pattern demanded for the `_xcsh_models=$(cat <<EOF ... EOF)` block around line 226.

### Log Output (last ~40 lines relevant)

```
##[group]SHELL_SHFMT
[ERROR]   Errors found in SHELL_SHFMT
[ERROR]   Stdout contents for SHELL_SHFMT:
diff /github/workspace/entrypoint.sh.orig /github/workspace/entrypoint.sh
--- /github/workspace/entrypoint.sh.orig
+++ /github/workspace/entrypoint.sh
@@ -186,30 +186,31 @@
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _pi_models=$(cat <<EOF
+  _pi_models=$(
+    cat <<EOF
 ...
 EOF
-)
+  )
  mkdir -p "$HOME/.pi/agent" "$HOME/.omp/agent"
@@ -226,7 +227,8 @@
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _xcsh_models=$(cat <<EOF
+  _xcsh_models=$(
+    cat <<EOF
 ...
 EOF
-)
+  )
[ERROR]   Super-linter detected linting errors
```

### Action Required

Apply shfmt's requested reformat to the two new heredoc blocks in `entrypoint.sh` (open the `$(` and close `)` on their own indented lines, with `cat <<EOF` on a separate line). Re-run the local pre-commit gate (it will still pass), then push a new commit to the PR branch.

Optional hardening follow-up: align the local pre-commit shell-script hook with super-linter's shfmt flags so this class of drift is caught before push.

--- a/crush-config/crush.json
+++ b/crush-config/crush.json
@@ -1,7 +1,30 @@
 {
   "$schema": "https://charm.land/crush.json",
+  "options": {
+    "disable_provider_auto_update": true
+  },
+  "providers": {
+    "anthropic": {
+      "base_url": "__CRUSH_BASE_URL__",
+      "api_key": "$LITELLM_API_KEY",
+      "models": [
+        {
+          "id": "pd-claude-opus-4-7",
+          "name": "Claude Opus 4.7 (PD)",
+          "cost_per_1m_in": 5,
+          "cost_per_1m_out": 25,
+          "cost_per_1m_in_cached": 6.25,
+          "cost_per_1m_out_cached": 0.5,
+          "context_window": 1000000,
+          "default_max_tokens": 128000,
+          "can_reason": true,
+          "supports_attachments": true
+        }
+      ]
+    }
+  },
   "models": {
     "large": { "model": "pd-claude-opus-4-7", "provider": "anthropic" },
-    "small": { "model": "claude-haiku-4-5", "provider": "anthropic" }
+    "small": { "model": "pd-claude-opus-4-7", "provider": "anthropic" }
   }
 }

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -174,28 +174,116 @@ if [ -n "$LITELLM_BASE_URL" ]; then
 fi
 
 # ============================================================
-# Pi & Oh-My-Pi — write models.json with LiteLLM proxy base URL.
-# Neither tool reads ANTHROPIC_BASE_URL from the environment;
-# they require a models.json provider override instead.
+# Pi & Oh-My-Pi — write models.json registering pd-claude-opus-4-7
+# as a custom model under the anthropic provider.
+#
+# Pi-ai (both @mariozechner/pi-ai and @oh-my-pi/pi-ai forks) keep a
+# built-in Anthropic model registry that does not include the custom
+# proxy ID `pd-claude-opus-4-7`. Without an explicit entry, the
+# resolver treats the ID as unknown, falls back to a built-in name
+# (e.g. `claude-opus-4-7` with no prefix), rewrites settings.json,
+# and the proxy rejects the stripped ID with 400. Declaring the
+# model here makes the resolver exact-match and stops the rewrite.
 # ============================================================
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _pi_models='{"providers":{"anthropic":{"baseUrl":"'"${LITELLM_BASE_URL}/anthropic"'"}}}'
+  _pi_models=$(cat <<EOF
+{
+  "providers": {
+    "anthropic": {
+      "baseUrl": "${LITELLM_BASE_URL}/anthropic",
+      "apiKey": "LITELLM_API_KEY",
+      "api": "anthropic-messages",
+      "models": [
+        {
+          "id": "pd-claude-opus-4-7",
+          "name": "Claude Opus 4.7 (PD)",
+          "api": "anthropic-messages",
+          "reasoning": true,
+          "input": ["text", "image"],
+          "cost": {"input": 5, "output": 25, "cacheRead": 0.5, "cacheWrite": 6.25},
+          "contextWindow": 1000000,
+          "maxTokens": 128000
+        }
+      ]
+    }
+  }
+}
+EOF
+)
   mkdir -p "$HOME/.pi/agent" "$HOME/.omp/agent"
   printf '%s\n' "$_pi_models" >"$HOME/.pi/agent/models.json"
   printf '%s\n' "$_pi_models" >"$HOME/.omp/agent/models.json"
+  # Reset pi's defaultModel on every boot — pi-ai's fallback path
+  # overwrites this file with the stripped ID; without this reset
+  # the corruption survives container restart when $HOME is mounted.
+  printf '{"defaultProvider":"anthropic","defaultModel":"pd-claude-opus-4-7"}\n' \
+    >"$HOME/.pi/agent/settings.json"
   unset _pi_models
 fi
 
 # ============================================================
-# xcsh — write models.json + models.yml with LiteLLM proxy base URL.
-# xcsh reads provider routing from ~/.xcsh/agent/models.json.
+# xcsh — write models.json + models.yml registering the custom
+# pd-claude-opus-4-7 model (same rationale as pi/omp above).
 # ============================================================
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _xcsh_models='{"providers":{"anthropic":{"baseUrl":"'"${LITELLM_BASE_URL}/anthropic"'"}}}'
+  _xcsh_models=$(cat <<EOF
+{
+  "providers": {
+    "anthropic": {
+      "baseUrl": "${LITELLM_BASE_URL}/anthropic",
+      "apiKey": "LITELLM_API_KEY",
+      "api": "anthropic-messages",
+      "models": [
+        {
+          "id": "pd-claude-opus-4-7",
+          "name": "Claude Opus 4.7 (PD)",
+          "api": "anthropic-messages",
+          "reasoning": true,
+          "input": ["text", "image"],
+          "cost": {"input": 5, "output": 25, "cacheRead": 0.5, "cacheWrite": 6.25},
+          "contextWindow": 1000000,
+          "maxTokens": 128000
+        }
+      ]
+    },
+    "litellm": {
+      "baseUrl": "${LITELLM_BASE_URL}/api/v1",
+      "apiKey": "LITELLM_API_KEY",
+      "api": "openai-completions"
+    }
+  }
+}
+EOF
+)
   mkdir -p "$HOME/.xcsh/agent"
   printf '%s\n' "$_xcsh_models" >"$HOME/.xcsh/agent/models.json"
-  printf 'providers: \n  anthropic: \n    baseUrl: %s\n' "${LITELLM_BASE_URL}/anthropic" \
-    >"$HOME/.xcsh/agent/models.yml"
+  cat >"$HOME/.xcsh/agent/models.yml" <<EOF
+configVersion: 2
+providers:
+  anthropic:
+    baseUrl: "${LITELLM_BASE_URL}/anthropic"
+    apiKey: LITELLM_API_KEY
+    api: anthropic-messages
+    models:
+      - id: pd-claude-opus-4-7
+        name: "Claude Opus 4.7 (PD)"
+        api: anthropic-messages
+        reasoning: true
+        input: [text, image]
+        cost:
+          input: 5
+          output: 25
+          cacheRead: 0.5
+          cacheWrite: 6.25
+        contextWindow: 1000000
+        maxTokens: 128000
+  litellm:
+    baseUrl: "${LITELLM_BASE_URL}/api/v1"
+    apiKey: LITELLM_API_KEY
+    api: openai-completions
+    discovery:
+      type: openai-compat
+EOF
   unset _xcsh_models
 fi
 
@@ -213,11 +301,49 @@ if [ -n "$LITELLM_BASE_URL" ]; then
 fi
 
 # ============================================================
-# Crush — route Anthropic provider through LiteLLM proxy.
-# Crush reads ANTHROPIC_API_ENDPOINT for the Anthropic base URL.
+# Crush — render base URL placeholder into crush.json.
+#
+# The tracked crush-config/crush.json uses __CRUSH_BASE_URL__ as a
+# placeholder so no internal hostnames live in the public repo.
+# Resolve it at container start to the LiteLLM Anthropic passthrough.
+#
+# crush.json also carries `options.disable_provider_auto_update: true`
+# plus a full `providers.anthropic.models[]` entry for
+# pd-claude-opus-4-7. Without the disable flag crush re-fetches the
+# upstream Catwalk catalog on every invocation, wiping custom models.
+# Without the inline models[] entry crush's embedded catalog has no
+# pd-claude-opus-4-7 and the proxy rejects the fallback ID.
 # ============================================================
 if [ -n "$LITELLM_BASE_URL" ]; then
+  _crush_config="$HOME/.config/crush/crush.json"
+  if [ -f "$_crush_config" ]; then
+    sed -i "s|__CRUSH_BASE_URL__|${LITELLM_BASE_URL}/anthropic|g" "$_crush_config"
+  fi
+  unset _crush_config
   export ANTHROPIC_API_ENDPOINT="${LITELLM_BASE_URL}/anthropic"
+fi
+
+# ============================================================
+# Proxy sanity probe.
+#
+# After all per-tool configs are rendered, POST a 1-token request
+# to the LiteLLM Anthropic passthrough to confirm pd-claude-opus-4-7
+# is accepted. Warns (non-fatal) on any non-200 response so the next
+# regression surfaces immediately instead of at first tool invocation.
+# ============================================================
+if [ -n "$LITELLM_BASE_URL" ] && [ -n "$LITELLM_API_KEY" ] && command -v curl >/dev/null 2>&1; then
+  _probe_status=$(curl -sS -o /dev/null -w '%{http_code}' \
+    -X POST "${LITELLM_BASE_URL}/anthropic/v1/messages" \
+    -H "x-api-key: ${LITELLM_API_KEY}" \
+    -H 'content-type: application/json' \
+    -H 'anthropic-version: 2023-06-01' \
+    -d '{"model":"pd-claude-opus-4-7","max_tokens":4,"messages":[{"role":"user","content":"hi"}]}' \
+    2>/dev/null || true)
+  if [ "$_probe_status" != "200" ]; then
+    printf 'WARN: LiteLLM proxy probe for pd-claude-opus-4-7 returned %s (expected 200)\n' \
+      "${_probe_status:-no-response}" >&2
+  fi
+  unset _probe_status
 fi
 
 # Re-sync Codex agents from Claude Code plugins (catches plugin updates)

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -186,7 +186,8 @@ fi
 # model here makes the resolver exact-match and stops the rewrite.
 # ============================================================
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _pi_models=$(cat <<EOF
+  _pi_models=$(
+    cat <<EOF
 {
   "providers": {
     "anthropic": {
@@ -209,7 +210,7 @@ if [ -n "$LITELLM_BASE_URL" ]; then
   }
 }
 EOF
-)
+  )
   mkdir -p "$HOME/.pi/agent" "$HOME/.omp/agent"
   printf '%s\n' "$_pi_models" >"$HOME/.pi/agent/models.json"
   printf '%s\n' "$_pi_models" >"$HOME/.omp/agent/models.json"
@@ -226,7 +227,8 @@ fi
 # pd-claude-opus-4-7 model (same rationale as pi/omp above).
 # ============================================================
 if [ -n "$LITELLM_BASE_URL" ]; then
-  _xcsh_models=$(cat <<EOF
+  _xcsh_models=$(
+    cat <<EOF
 {
   "providers": {
     "anthropic": {
@@ -254,7 +256,7 @@ if [ -n "$LITELLM_BASE_URL" ]; then
   }
 }
 EOF
-)
+  )
   mkdir -p "$HOME/.xcsh/agent"
   printf '%s\n' "$_xcsh_models" >"$HOME/.xcsh/agent/models.json"
   cat >"$HOME/.xcsh/agent/models.yml" <<EOF

--- a/opencode-config/opencode.json
+++ b/opencode-config/opencode.json
@@ -10,7 +10,7 @@
       },
       "models": {
         "pd-claude-opus-4-7": {
-          "name": "Claude Opus 4.6",
+          "name": "Claude Opus 4.7 (PD)",
           "modalities": {
             "input": ["text", "image"],
             "output": ["text"]

